crates/bevy_input/src/gamepad.rs

```diff
@@ -1092,137 +937,310 @@ impl Default for ButtonAxisSettings {
 }
 
 impl ButtonAxisSettings {
-    /// Filters the `new_value` according to the specified settings.
+    /// Clamps the `raw_value` according to the specified settings.
     ///
-    /// If the `new_value` is:
+    /// If the `raw_value` is:
     /// - lower than or equal to `low` it will be rounded to 0.0.
     /// - higher than or equal to `high` it will be rounded to 1.0.
     /// - Otherwise it will not be rounded.
+    fn clamp(&self, raw_value: f32) -> f32 {
+        if raw_value <= self.low {
+            return 0.0;
+        }
+        if raw_value >= self.high {
+            return 1.0;
+        }
+
+        raw_value
+    }
+
+    /// Determines whether the change from an `old_value` to a `new_value` should
+    /// be registered as a change event, according to the specified settings.
+    fn should_register_change(&self, new_value: f32, old_value: Option<f32>) -> bool {
+        if old_value.is_none() {
+            return true;
+        }
+
+        f32::abs(new_value - old_value.unwrap()) > self.threshold
+    }
+
+    /// Filters the `new_value` based on the `old_value`, according to the `ButtonAxisSettings`.
     ///
-    /// If the difference between the calculated value and the `old_value` is lower or
-    /// equal to the `threshold`, [`None`] will be returned.
+    /// Returns the [clamped](`ButtonAxisSettings::clamp`) `new_value` if the change exceeds the settings threshold,
+    /// and `None` otherwise.
+    #[allow(unused)]
     fn filter(&self, new_value: f32, old_value: Option<f32>) -> Option<f32> {
```

Comment:
`filter` isn't used in the runtime code, hence `#[allow(unused)]`, but it's used and is quite handy for the doc tests. We could create a standalone `filter` just for the tests but keeping it keeps the `ButtonAxisSettings` and `AxisSettings` APIs consistent.

Reply:
You should make it pub rather than allowing unused here IMO.